### ISSUE TITLE
Look at further ways to use Array methods where possible

### ISSUE THREAD

**Issue by muffinresearch** [mozilla/addons-linter#1456]
In updating the eslint config it came up that there are a lot of loops and the new lint config we've avoided the use of `for-of` / `for-in`.

Generally the approach taken was to do the smallest amount of changes - however, there's definitely likely to be some further opportunities to improve the loops and make more use of `map` and `reduce` etc.

From the [airbnb style-guide](https://github.com/airbnb/javascript/blob/master/README.md#iterators-and-generators):

> Use map() / every() / filter() / find() / findIndex() / reduce() / some() / ... to iterate over arrays, and Object.keys() / Object.values() / Object.entries() to produce arrays so you can iterate over objects.

```javascript
const numbers = [1, 2, 3, 4, 5];

// bad
let sum = 0;
for (let num of numbers) {
  sum += num;
}
sum === 15;

// good
let sum = 0;
numbers.forEach((num) => {
  sum += num;
});
sum === 15;

// best (use the functional force)
const sum = numbers.reduce((total, num) => total + num, 0);
sum === 15;

// bad
const increasedByOne = [];
for (let i = 0; i < numbers.length; i++) {
  increasedByOne.push(numbers[i] + 1);
}

// good
const increasedByOne = [];
numbers.forEach((num) => {
  increasedByOne.push(num + 1);
});

// best (keeping it functional)
const increasedByOne = numbers.map(num => num + 1);
```

**Comment by EnTeQuAk** [comment]
All these feel like micro optimizations with no real-world use. I do get using `map` or `reduce` for these counting tasks but that's not what most of our loops look like. It also makes loops and usually very basic code enormously hard to read. I would say that I am a relatively experienced developer but whenever I'm faced with map, reduce and many other functional constructs that transform data in a very in-transparent way it's hard to follow along and I spend more time trying to understand simple constructs that I shouldn't even care about.

Let's please not use functional statements just for the sake of using them and keep code-quality and readability high, the linter already suffers quite hard from it's massive amount of code that is only there because of promises, pipelines and stuff which makes it hard to follow code-paths.

**Comment by muffinresearch** [comment]
@EnTeQuAk I definitely don't want to do anything for the sake of it. However one of the reasons people care about purity is the difference between code which modifies versus creating a new list which TBH is probably a lot more important in a react code-base than it is in something like the linter.

Maybe if we ever review this stuff we might decide there's nothing to improve upon and if that's the case then I'm fine with that, I only wanted to flag this since I didn't get too deep into working out if there is room for improvement.
